api/src/main/java/module-info.java

```diff
@@ -1109,6 +1111,37 @@
  * {@code @Transactional} is called in an environment where both Jakarta Transactions and
  * Jakarta CDI are available, the semantics of this annotation are observed during execution
  * of the repository operation.</p>
+ *
+ * <h2>Jakarta Concurrency</h2>
+ *
+ * <p>When Jakarta Concurrency is available, an asynchronous repository method may be
+ * annotated with {@code jakarta.enterprise.concurrent.Asynchronous} to cause the
+ * method to run asynchronously to the method invoker, as outlined by the section
+ * titled Asynchronous Methods in the Jakarta Concurrency specification.
+ * A {@code CompletionStage} return type must be parameterized with a type that
+ * would be a valid return type for a non-asynchronous repository method.</p>
```

Comment:
```suggestion
 * The return type must be {@code CompletionStage<R>} where {@code R} is a type 
 * that would be a valid return type for a non-asynchronous repository method.</p>
```

Reply:
`void` is also an acceptable return type (and currently works), so I'll plan to add that in after committing this suggestion.